### PR TITLE
Integrate TEC Frontend with New Backend Endpoints

### PR DESCRIPTION
**Pull request by airaagrawal1**
### Summary 
This PR integrates the TEC frontend with the new backend endpoints. Mock data was removed and live data is being used on the frontend now. 

### Notion/Figma Link 
https://www.notion.so/cornelldti/Idol-SP23-e65533742dca4f898581cc5c496e821e?p=3c685dbe13ed416e90cc197786a17e9a&pm=s

### Test Plan 
Manually tested all functionality on member and admin sides and confirmed expected outputs. Checked dev database to ensure data in DB corresponded to data being displayed on the frontend.

### CODE REVIEW

File: backend/src/API/teamEventsAPI.ts [hunk 1]
```diff
@@ -72,8 +72,7 @@ export const requestTeamEventCredit = async (
       `User with email ${user.email} cannot request team event credit for another member, ${request.member.email}.`
     );
   }
-  const updatedteamEvent = { ...request, pending: true };
-  await teamEventAttendanceDao.createTeamEventAttendance(updatedteamEvent);
+  await teamEventAttendanceDao.createTeamEventAttendance(request);
```
Comment (henry-li-06) [inline backend/src/API/teamEventsAPI.ts:75]: What's the purpose of this change?

File: backend/src/dao/TeamEventAttendanceDao.ts [hunk 2]
```diff
@@ -46,7 +46,6 @@ export default class TeamEventAttendanceDao extends BaseDao<
   ): Promise<TeamEventAttendance> {
     const teamEventAttendanceWithUUID = {
       ...teamEventAttendance,
-      pending: true,
```
Comment (henry-li-06) [inline backend/src/dao/TeamEventAttendanceDao.ts:49]: Why are we getting rid of this?

File: backend/src/dao/TeamEventsDao.ts [hunk 3]
```diff
@@ -60,6 +60,6 @@ export default class TeamEventsDao {
     const docRefs = await teamEventsCollection
       .select('name', 'date', 'numCredits', 'hasHours', 'uuid')
       .get();
-    return docRefs.docs.map((doc) => doc.data() as TeamEventInfo);
+    return Promise.all(docRefs.docs.map(async (doc) => doc.data() as TeamEventInfo));
```
Comment (henry-li-06) [inline backend/src/dao/TeamEventsDao.ts:63]: I don't believe `doc.data()` returns a promise. 
image

File: frontend/src/API/TeamEventsAPI.ts [hunk 4]
```diff
@@ -42,7 +42,8 @@ export class TeamEventsAPI {
         });
         return [];
       }
-      return val.allTeamEventInfo as TeamEventInfo[];
+      const events = val.allTeamEventInfo as TeamEventInfo[];
+      return events;
```
Comment (henry-li-06) [inline frontend/src/API/TeamEventsAPI.ts:46]: It's unnecessary to break this up into 2 lines.

File: frontend/src/components/Admin/TeamEvent/TeamEventCreditReview.tsx [hunk 5]
```diff
@@ -39,21 +39,20 @@ const TeamEventCreditReview = (props: {
...
+          contentMsg: 'The team event attendance was successfully rejected!'
         });
-    });
+      })
+      .catch((error) => {
+        Emitters.generalError.emit({
+          headerMsg: "Couldn't reject the team event attendance!",
+          contentMsg: error
+        });
+      });
```
Comment (henry-li-06) [inline frontend/src/components/Admin/TeamEvent/TeamEventCreditReview.tsx:55]: I think that it would be better to make sure that the attendance doc gets deleted first before deleting the proof image. That way if something goes wrong with deleting the attendance doc, we haven't already accidentally deleted the attendance doc. We're better off having deleting the attendance doc and having the deletion of the image fail for whatever reason.

File: frontend/src/components/Forms/TeamEventCreditsForm/TeamEventsCreditDashboard.tsx [hunk 6]
```diff
@@ -68,11 +68,11 @@ const TeamEventCreditDashboard = (props: {
     const { attendanceList } = props;
     return (
       <Card.Group>
-        {attendanceList.map((attendance) => {
+        {attendanceList.map((attendance, i) => {
           const teamEvent = allTEC.find((tec) => tec.uuid === attendance.eventUuid);
           if (teamEvent !== undefined) {
             return (
-              <Card>
+              <Card key={i}>
```
Comment (henry-li-06) [inline frontend/src/components/Forms/TeamEventCreditsForm/TeamEventsCreditDashboard.tsx:75]: Probably better to use `attendance.uuid` for the `key` prop.

File: backend/src/API/teamEventsAPI.ts (hunk 1 above)
Comment (airaagrawal1) [reply backend/src/API/teamEventsAPI.ts:75]: The team event request from the frontend should already have pending set to true by default, so this would be redundant.

File: backend/src/dao/TeamEventAttendanceDao.ts (hunk 2 above)
Comment (airaagrawal1) [reply backend/src/dao/TeamEventAttendanceDao.ts:49]: Same comment as above.

File: backend/src/API/teamEventsAPI.ts (hunk 1 above)
Comment (henry-li-06) [reply backend/src/API/teamEventsAPI.ts:75]: Backend shouldn't assume that data from the frontend is "correct"

File: frontend/src/API/TeamEventsAPI.ts (hunk 4 above)
Comment (airaagrawal1) [reply frontend/src/API/TeamEventsAPI.ts:46]: My bad, was testing some stuff.

File: backend/src/API/teamEventsAPI.ts (hunk 1 above)
Comment (airaagrawal1) [reply backend/src/API/teamEventsAPI.ts:75]: 👍🏽

File: frontend/src/components/Forms/TeamEventCreditsForm/TeamEventsCreditDashboard.tsx (hunk 6 above)
Comment (airaagrawal1) [reply frontend/src/components/Forms/TeamEventCreditsForm/TeamEventsCreditDashboard.tsx:75]: Makes sense

File: backend/src/dao/TeamEventsDao.ts (hunk 3 above)
Comment (airaagrawal1) [reply backend/src/dao/TeamEventsDao.ts:63]: I thought this change had fixed an error I was getting, but it seems like both ways work. I will change it back.

File: frontend/src/components/Admin/TeamEvent/TeamEventCreditReview.tsx (hunk 5 above)
Comment (airaagrawal1) [reply frontend/src/components/Admin/TeamEvent/TeamEventCreditReview.tsx:55]: Makes sense

**Review by henry-li-06 — APPROVED** [review]
Nice work!
